subsys/pm/device_runtime.c

```diff
@@ -44,7 +44,11 @@ static int runtime_suspend(const struct device *dev, bool async)
 	if (k_is_pre_kernel()) {
 		async = false;
 	} else {
-		(void)k_mutex_lock(&pm->lock, K_FOREVER);
+		ret = k_mutex_lock(&pm->lock,
+				   k_is_in_isr() ? K_NO_WAIT : K_FOREVER);
```

Comment:
https://github.com/zephyrproject-rtos/zephyr/blob/89c7745c0148e54999f31cf9c8ae034cd18fd2f1/kernel/mutex.c#L101

And what about this?

Reply:
I forgot you can't use mutexes (non-blocking) in ISRs. So I guess we have a big problem here, since PM subsys depends on condition variables that in turn depend on mutexes...

Reply:
this is now fixed thanks to https://github.com/zephyrproject-rtos/zephyr/pull/52834